cmd/earthly/build_cmd.go

```diff
@@ -573,11 +606,19 @@ func (app *earthlyApp) actionBuildImp(cliCtx *cli.Context, flagArgs, nonFlagArgs
 	if app.logstream && doLogstreamUpload && !app.logbusSetup.LogStreamerStarted() {
 		app.console.ColorPrintf(color.New(color.FgHiYellow), "Streaming logs to %s\n\n", logstreamURL)
 	}
+
 	_, err = b.BuildTarget(cliCtx.Context, target, buildOpts)
 	if err != nil {
 		return errors.Wrap(err, "build target")
 	}
 
+	if app.skipBuildkit && targetHash != nil {
```

Comment:
skipDB is guarenteed to be created during the first time when we check `app.skipBuildkit`.

We could do a ```
if skipDB != nil {
  panic("skipDB should have been created")
}
```

if we're really worried about it, but it seems overkill.

Reply:
I guess I'm just worried about it because there are a few `if` blocks that don't end in a `return` - although, to your point, the `targetHash` _should_ be `nil` if `skipDB` is `nil` here. I just worry about nil pointer dereferences if more cases are added where `skipDB == nil` - although I can't think of reasonable cases where `targetHash != nil && skipDB == nil`, even in the distant future.